server/segment_rpc.go

```diff
@@ -295,6 +296,31 @@ func makeFfmpegVideoProfiles(protoProfiles []*net.VideoProfile) ([]ffmpeg.VideoP
 	return profiles, nil
 }
 
+func makeNetDetectData(ffmpegDetectData []ffmpeg.DetectData) []*net.DetectData {
+	netDataList := []*net.DetectData{}
+	for _, data := range ffmpegDetectData {
+		var netData *net.DetectData
+		switch data.Type() {
+		case ffmpeg.SceneClassification:
+			d := data.(ffmpeg.SceneClassificationData)
+			netClasses := make(map[uint32]*net.Fraction)
+			for classID, prob := range d {
+				netClasses[uint32(classID)] = &net.Fraction{
```

Comment:
```suggestion
				netClasses[uint32(classID)] = &net.Fraction{Num: 0, Denom: 1}
                                  rat := new(big.Rat).SetFloat64(prob)
                                  if rat != nil {
                                      netClasses[uint32(classID)].Num = rat.Num()
                                      netClasses[uint32(classID)].Denom = rat.Denom()
                                  }
```

^ Suggestion for converting the float64 probability into a fraction that doesn't require hardcoded conversation constants.

Reply:
Hmm `rat.Num()` returns a `big.Int` which might not necessarily fit our `net.Fraction.Num` which is a `uint32`. https://play.golang.org/p/XLYKIR31Lvt

Even if we convert `big.Int` using the `Uint64` method, there is no guarantee that an arbitrary float we get from the C code will not overflow it -> the docs suggest runtime checks if it overflows via https://golang.org/pkg/math/big/#Int.IsUint64

Maybe we can stick with a hardcoded constant for now? At least till we figure out how to handle floats/fractions consistently across C->Go->Protobuf->JSON.

I will increase the constant from `1000` to `1e6` to retain more precision.

Reply:
Good catch.

I suspect there might be easier ways to pass a float across the wire...Let's revisit this.